### PR TITLE
feat(tcgc): bump to 0.41.5

### LINKED ISSUE #4618: Projected version support Implementation
Should be as simple as move to TCGC 0.41.4+

### PR DESCRIPTION
**Pull request by archerzz**
# Description

- bump tcgc to 0.41.5
- restore a unit test previously disabled due to tcgc bug
- add `api-version` option property
- add test case for version projection

resolve #4618

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by archerzz** [comment]
Preview gen: https://github.com/Azure/azure-sdk-for-net/pull/43668

File: test/AutoRest.TestServerLowLevel.Tests/dpg-version.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using TypeSpec.Versioning.Latest.Models;
+using NUnit.Framework;
+using System.Linq;
+using System.Collections.Generic;
+using Resource = TypeSpec.Versioning.Latest.Models.Resource;
+using System.Threading;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class DpgVersionTest : TestServerLowLevelTestBase
```
Comment (pshao25) [inline test/AutoRest.TestServerLowLevel.Tests/dpg-version.cs:17]: If you are not calling test server, there is no need to inherit TestServerLowLevelTestBase

Comment (archerzz) [reply test/AutoRest.TestServerLowLevel.Tests/dpg-version.cs]: removed the base class

File: src/TypeSpec.Extension/Emitter.Csharp/src/options.ts [hunk 2]
```diff
@@ -120,6 +122,7 @@ export const NetEmitterOptionsSchema: JSONSchemaType<NetEmitterOptions> = {
 };
 
 const defaultOptions = {
+    "api-version": "latest",
```
Comment (lirenhe) [inline src/TypeSpec.Extension/Emitter.Csharp/src/options.ts:125]: I didn't find the change in emitter.ts, are we able to load it from configuration?

Comment (archerzz) [reply src/TypeSpec.Extension/Emitter.Csharp/src/options.ts:125]: > I didn't find the change in emitter.ts, are we able to load it from configuration?

Yes. It's `api-version` in `tspconfig.yaml`

**Review by lirenhe — APPROVED** [review]

**Pull request merged**